lib/Test/Nginx/Util.pm

```diff
@@ -2703,4 +2763,78 @@ sub use_http2 ($) {
     return undef;
 }
 
+
+sub use_http3 ($) {
+    my $block = shift;
+    my $cached = $block->test_nginx_enabled_http3;
+
+    if (defined $cached) {
+        return $cached;
+    }
+
+    if (defined $block->http3) {
+        if ($block->raw_request) {
+            bail_out("cannot use --- http3 with --- raw_request");
+        }
+
+        if ($block->pipelined_requests) {
+            bail_out("cannot use --- http3 with --- pipelined_requests");
+        }
+
+        if (defined $block->http2) {
+            bail_out("cannot use --- http3 with --- http2");
+        }
+
+        $block->set_value("test_nginx_enabled_http3", 1);
+
+        if (!$LoadedIPCRun) {
+            require IPC::Run;
+            $LoadedIPCRun = 1;
+        }
+        return 1;
+    }
+
+    if ($UseHttp3) {
+        if ($block->raw_request) {
+            warn "WARNING: ", $block->name, " - using raw_request HTTP/2, will not use HTTP/3\n";
+            $block->set_value("test_nginx_enabled_http3", 0);
+            return undef;
+        }
+
+        if ($block->pipelined_requests) {
+            warn "WARNING: ", $block->name, " - using pipelined_requests, will not use HTTP/3\n";
+            $block->set_value("test_nginx_enabled_http3", 0);
+            return undef;
+        }
+
+        if (!defined $block->request) {
+            $block->set_value("test_nginx_enabled_http3", 0);
+            return undef;
+        }
+
+        if (!ref $block->request && $block->request =~ m{HTTP/1\.0}s) {
+            warn "WARNING: ", $block->name, " - explicitly rquires HTTP 1.0, so will not use HTTP/3\n";
+            $block->set_value("test_nginx_enabled_http3", 0);
+            return undef;
+        }
+
+        if (defined $block->http2) {
+            warn "WARNING: ", $block->name, " - explicitly rquires HTTP/2, so will not use HTTP/3\n";
+            $block->set_value("test_nginx_enabled_http3", 0);
+            return undef;
+        }
+
+        $block->set_value("test_nginx_enabled_http3", 1);
+
+        if (!$LoadedIPCRun) {
+            require IPC::Run;
+            $LoadedIPCRun = 1;
+        }
+        return 1;
+    }
+
+    $block->set_value("test_nginx_enabled_http3", 0);
```

Comment:
set_value to test_nginx_enabled_http3 should be `1` here

Reply:
0 is right.
use_http3 is used to test if should use http3 for the test case, not to enable http3.